File: map-extraction-service - claude/services/map-service/app/utils/geo_utils.py
def calculate_area(rectangle):
    """Calculate the area of a rectangle given its coordinates."""
    x1, y1, x2, y2 = rectangle
    return abs((x2 - x1) * (y2 - y1))

def is_point_in_rectangle(point, rectangle):
    """Check if a point is inside a given rectangle."""
    x, y = point
    x1, y1, x2, y2 = rectangle
    return x1 <= x <= x2 and y1 <= y <= y2

def rectangle_to_geojson(rectangle):
    """Convert a rectangle defined by its coordinates to GeoJSON format."""
    x1, y1, x2, y2 = rectangle
    coordinates = [
        [x1, y1],
        [x2, y1],
        [x2, y2],
        [x1, y2],
        [x1, y1]  # Closing the polygon
    ]
    return {
        "type": "Feature",
        "geometry": {
            "type": "Polygon",
            "coordinates": [coordinates]
        },
        "properties": {}
    }

File: geo-building-extractor/frontend/src/utils/geoUtils.js
// This file provides utility functions for geographic calculations.

export const calculateArea = (rectangle) => {
    const { northEast, southWest } = rectangle;
    const width = northEast.lng - southWest.lng;
    const height = northEast.lat - southWest.lat;
    return Math.abs(width * height);
};

export const rectangleToGeoJSON = (rectangle) => {
    const { northEast, southWest } = rectangle;
    return {
        type: "Feature",
        geometry: {
            type: "Polygon",
            coordinates: [[
                [southWest.lng, southWest.lat],
                [northEast.lng, southWest.lat],
                [northEast.lng, northEast.lat],
                [southWest.lng, northEast.lat],
                [southWest.lng, southWest.lat]
            ]]
        },
        properties: {}
    };
};

export const isWithinFreeLimit = (area) => {
    const freeLimit = 1000000; // 1 km² in square meters
    return area <= freeLimit;
};